Add grouping of multichain assets to the api balances and responses

yabirgb opened the pull request:
This PR addresses the problem of presenting to the frontend the information about assets in multiple chains.

For the balances it follows one schema like this one

```python
{
        'assets': {
            'USD': {'amount': '2', 'usd_value': '2'},
            'ETH': {'amount': '3', 'usd_value': '900'},
            'eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48': {
                'amount': '5',
                'usd_value': '5',
                'breakdown': {
                    'eip155:1/erc20:0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48': {
                        'amount': '2',
                        'usd_value': '2'
                    },
                    'eip155:10/erc20:0x7F5c764cBc14f9669B88837ca1490cCa17c31607': {
                        'amount': '3',
                        'usd_value': '3',
                    }
                },
            }
        },
        'liabilities': {
            ethaddress_to_identifier('0x6B175474E89094C44Da98b954EedeAC495271d0F'): {'amount': '5', 'usd_value': '5.1'},
            'ETH': {'amount': '0.5', 'usd_value': '150'},
        },
    }
```

and for the `/assets/mappings` we include now if the asset is part of a collection and then provide the collection details. Check the docs for a detailed example

**This PR includes a new update in the globaldb adding the collections**. The file with the groups was created using this script https://gist.github.com/yabirgb/258f29c33c59c88fc82cc8fddbb09bdd

yabirgb commented on rotkehlchen/accounting/structures/balance.py:
```diff
@@ -18,6 +20,67 @@ class BalanceType(DBEnumMixIn):
+            crypto_asset = asset.resolve_to_crypto_asset()
+            if isinstance(crypto_asset, EvmToken) is True:
+                sorting_key = crypto_asset.chain_id  # type: ignore  # noqa: E501 mypy fails to see that this could be EvmToken
+            else:
+                sorting_key = 0
```
```suggestion
            else:
                # we need to do this to avoid issues in the case of collections for non evm tokens
                sorting_key = 0
```

yabirgb commented on rotkehlchen/tests/api/test_blockchain.py:
```diff
@@ -219,7 +219,7 @@ def test_query_blockchain_balances(
         else:
             outcome = assert_proper_response_with_result(response)
-
+    assert False
```
wtf dude

yabirgb reviewed (commented):
You left an empty file. Seriously take care when pushing

nebolax commented on docs/api.rst:
```diff
@@ -2782,69 +2782,129 @@ Querying onchain balances
+                  "eip155:1/erc20:0x6B175474E89094C44Da98b954EedeAC495271d0F":{
+                    "amount":"15",
+                    "usd_value":"15.21",
+                    "breakdown":{
```
Question. Why a breakdown is needed if user's owned asset belongs only to one chain?

yabirgb commented on rotkehlchen/globaldb/handler.py:
```diff
@@ -430,8 +446,14 @@ def get_assets_mappings(identifiers: list[str]) -> dict[str, dict]:
-
-        return result
+                if entry[6] is not None:
+                    result[entry[0]].update({'collection_id': str(entry[6])})
```
Because numbers can't be keys to json

yabirgb commented on rotkehlchen/api/rest.py:
```diff
@@ -1396,13 +1396,17 @@ def query_list_of_all_assets(
     def get_assets_mappings(self, identifiers: list[str]) -> Response:
         try:
-            asset_mappings = GlobalDBHandler().get_assets_mappings(identifiers)
+            asset_mappings, asset_collections = GlobalDBHandler().get_assets_mappings(identifiers)
```
To avoid duplication in the information and so the frontend can group them later. So for each asset that forms a group we give their group id and then the groups names/symbols

yabirgb commented:
@nebolax thanks! We have changed a bit the approach and after your comment we have moved the grouping from the backend to the frontend to avoid the duplications. We only are adding the collections to the database here and the assets groups in the reponse for the assets

nebolax reviewed (commented):
Just a few comments

nebolax commented on rotkehlchen/globaldb/handler.py:
```diff
@@ -430,8 +447,14 @@ def get_assets_mappings(identifiers: list[str]) -> dict[str, dict]:
+                if entry[6] is not None:
+                    result[entry[0]].update({'collection_id': str(entry[6])})
+                    if entry[6] not in asset_collections:
+                        asset_collections[str(entry[6])] = {
```
Discussed in discord. All good here. My fault -- misread the code

yabirgb replied:
It was a misunderstanding about how the collection worked

yabirgb commented on rotkehlchen/globaldb/handler.py:
```diff
@@ -1908,3 +1931,17 @@ def get_asset_type(identifier: str, use_packaged_db: bool = False) -> AssetType:
         return AssetType.deserialize_from_db(type_in_db[0])
+
+    @staticmethod
+    def get_assets_collections(asset_ids: list[str]) -> dict[str, int]:
```
True is not used. The usage was removed after updating the logic. Thansk

yabirgb commented on docs/api.rst:
```diff
@@ -2782,69 +2782,117 @@ Querying onchain balances
+                  "amount":"0.5",
+                  "usd_value":"3770.075"
+                },
+                "33hjmoU9XjEz8aLxf44FNGB8TdrLkAVBBo":{
```
I updated them because fixed the indentation and other stuff when changing the structure before but will undo that change

yabirgb commented on docs/api.rst:
```diff
@@ -3114,35 +3162,44 @@ Get asset identifiers mappings
+              "eip155:1/erc20:0xB6eD7644C69416d67B522e20bC294A9a9B405B31":{
+                  "name":"0xBitcoin",
+                  "symbol":"0xBTC",
+                  "asset_type":"evm token"
```
yes ups, seems that when formatting I delete it. Thanks

LefterisJP commented on docs/api.rst:
```diff
@@ -3114,35 +3114,45 @@ Get asset identifiers mappings
-                  "asset_type": "evm token"
+        "result":{
+            "assets":{
+              "eip155:1/erc20:0xB6eD7644C69416d67B522e20bC294A9a9B405B31":{
```
why remove the space between `:` and `{` ? This is how json is styled in this file . Same in the rest of the diff

LefterisJP commented on docs/api.rst:
```diff
@@ -3114,35 +3114,45 @@ Get asset identifiers mappings
+              "eip155:1/erc20:0xcC4eF9EEAF656aC1a2Ab886743E98e97E090ed38":{
+                  "name":"DigitalDevelopersFund",
+                  "symbol":"DDF",
+                  "chain_id":1,
```
why did you remove the styling here? Please undo and make it like the rest of the file with a space between `:` and the other stuff. In all of the json

LefterisJP commented on rotkehlchen/data/assets_collections.sql:
```diff
@@ -0,0 +1,726 @@
+INSERT INTO asset_collections(id, name, symbol) VALUES 
```
shouldn't we do both here and below `OR IGNORE` just to be safe?

yabirgb replied:
okey

LefterisJP commented on rotkehlchen/tests/unit/test_globaldb_upgrades.py:
```diff
@@ -29,19 +29,48 @@
-def _count_v2_v3_assets_inserted() -> int:
-    """Counts and returns how many assets are to be inserted by globaldb_v2_v3_assets.sql"""
-    assets_inserted_by_update = 0
+def _count_sql_file_sentences(file_name: str, skip_tables: int = 0):
```
shouln't it be skip_statements? As there is no guarantee two different statements don't touch the same table?

yabirgb replied:
Yes. I had one table per statmenent but makes more sense to rename it

LefterisJP commented on rotkehlchen/tests/unit/test_globaldb_upgrades.py:
```diff
@@ -29,19 +29,48 @@
+
+def _count_v2_v3_assets_inserted() -> int:
+    """Counts and returns how many assets are to be inserted by globaldb_v2_v3_assets.sql"""
+    return _count_sql_file_sentences('globaldb_v2_v3_assets.sql')
```
Why make all those different functions with just 1 other function call in the body? Keep it simple and just use the `count_sql_file_sententces` where you need. These functions are not needed

yabirgb replied:
It was to make it clearer but will change it and just call `_count_sql_file_sentences`

LefterisJP commented on rotkehlchen/tests/unit/test_globaldb_upgrades.py:
```diff
@@ -29,19 +29,33 @@
-    assets_inserted_by_update = 0
+def _count_sql_file_sentences(file_name: str, skip_statements: int = 0):
+    """
+    Count the sql lines in scripts used during upgrades. If the skip_tables argument is provided
```
Yabir my man I told you to rename it. That means also:

1. in the docstring
2. Inside the function. `skipped_tables` should be `skipped_statements`
